backends/symbolic-balancing/balancing-parse-input/Backend/JsonToData.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Backend.JsonToData where
+
+import           Cardano.Api
+import           Cardano.Api.Shelley  (PlutusScript (..), scriptDataFromJsonDetailedSchema,
+                                       shelleyPayAddrToPlutusPubKHash, toPlutusData)
+import           Control.Monad        (zipWithM)
+import           Data.Aeson           (decode, eitherDecode, parseJSON, (.:), (.:?))
+import qualified Data.Aeson.Key       as Key
+import qualified Data.Aeson.KeyMap    as KeyMap
+import           Data.Aeson.Types     as Aeson (Result (..), Value (..), parse, parseEither)
+import           Data.Bifunctor       (first)
+import qualified Data.ByteString.Lazy as BL
+import           Data.Either
+import           Data.Maybe           (Maybe (..), fromMaybe)
+import           Data.Text            (Text)
+import qualified Data.Text            as T
+import qualified Data.Text.Encoding   as TE
+import           PlutusLedgerApi.V3   as V3
+import           PlutusTx             (CompiledCode)
+import           Prelude              (String, const, maybe, return, show, ($), (++), (.))
+import           Text.Read            (readEither)
+
+-- Function to parse JSON and convert to a list of TxInInfo
+parseJsonToTxInInfoList :: [Maybe (CompiledCode a)] -> BL.ByteString -> Either String [TxInInfo]
```

Comment:
I think a similar parser should be in the Atlas 🤔

Reply:
Yes, with Atlas all of this should be simpler.